skimage/feature/peak.py

```diff
@@ -172,6 +173,20 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
 
     threshold_abs = threshold_abs if threshold_abs is not None else image.min()
 
+    if min_distance < 1:
+        warn("min_distance lower than 1 is deprecated. "
+             "In this case, peak_local_max acts as finding "
+             "image > max(threshold_abs, threshold_rel). In version 0.20 "
+             "this warning will be replaced by a ValueError.",
+             FutureWarning, stacklevel=2)
+
+    if footprint is not None and footprint.size == 1:
+        warn("footprint.size lower than 2 is deprecated. "
+             "In this cases, peak_local_max acts as finding "
```

Comment:
`"In this case (...)`

Reply:
Perhaps not intentionally, but previously an array-like footprint did work. (e.g. `footprint=[[0, 1, 0], [1, 1, 1], [0, 1, 0]]`), but now this would result in an AttributeError. The docstring does indicate that `footprint` should be an ndarray, though, so perhaps it is okay to leave it as is.

If we want to maintain array-like support, use something like:
```Python
if footprint is not None:
    footprint = np.asarray(footprint)
    if footprint.size == 1:
        warn(...
```

Reply:
Good catch, I will keep the `array-like` support :wink: